sfxclient/multitokensink.go

```diff
@@ -0,0 +1,515 @@
+package sfxclient
+
+import (
+	"fmt"
+	"hash"
+	"hash/fnv"
+	"net/http"
+	"sync"
+	"sync/atomic"
+	"time"
+
+	"github.com/signalfx/golib/datapoint"
+	"github.com/signalfx/golib/event"
+	"golang.org/x/net/context"
+)
+
+// ContextKey is a custom key type for context values
+type ContextKey string
+
+const (
+	// TokenCtxKey is a context key for tokens
+	TokenCtxKey ContextKey = TokenHeaderName
+)
+
+type msg struct {
+	token string
+	data  interface{}
```

Comment:
It allows me to have a common base class called Worker that creates a common input and Stop mechanism.

Reply:
So have each base class implement a method like emitStop() and and call it on an interface? or use signals instead of your emitting a stop datapoint. There are a variety of ways to do this and keep type safety.